File: Beginner/Game_Time_With_Minutes_1047/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	values := bufio.NewReader(os.Stdin)
	line, _ := values.ReadString('\n')
	line = strings.Trim(line, "\n")
	valuesSplitted := strings.Split(line, " ")
	fmt.Println(valuesSplitted[0])
	initialHour, _ := strconv.ParseInt(valuesSplitted[0], 10, 64)
	initialMinutes, _ := strconv.ParseInt(valuesSplitted[1], 10, 64)
	endHour, _ := strconv.ParseInt(valuesSplitted[2], 10, 64)
	endMinutes, _ := strconv.ParseInt(valuesSplitted[3], 10, 64)

	initialDuration := initialHour*60 + initialMinutes
	endDuration := endHour*60 + endMinutes

	totalMinutes := endDuration - initialDuration
	if totalMinutes <= 0 {
		totalMinutes = totalMinutes + 24*60
	}

	totalHours := totalMinutes / 60
	totalMinutes = totalMinutes % 60
	fmt.Printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)", totalHours, totalMinutes)
}

File: Beginner/Game_Time_With_Minutes_1047/main.py
data_input = [int(number) for number in input().split(" ")]
initial_hour = data_input[0]
initial_minutes = data_input[1]
end_hour = data_input[2]
end_minutes = data_input[3]

initial_time_minutes = initial_hour * 60 + initial_minutes
end_time_minutes = end_hour * 60 + end_minutes

total_minutes = end_time_minutes - initial_time_minutes

if total_minutes <= 0:
    total_minutes = total_minutes + 24 * 60

total_hours = total_minutes // 60
total_minutes = total_minutes % 60

print(f'O JOGO DUROU {total_hours} HORA(S) E {total_minutes} MINUTO(S)')
